### PR TITLE
ossutil: init at 1.7.6

### PR DESCRIPTION
**Pull request by jpetrucciani**
###### Motivation for this change

I'd like to be able to use the [`ossutil`](https://github.com/aliyun/ossutil/) cli tool for Alibaba Cloud's [Object Storage Service](https://www.alibabacloud.com/product/oss) (OSS) through nix!

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/ossutil/use_go_mod.patch [hunk 1]
```diff
@@ -0,0 +1,125 @@
+From 64067e979fb24ffb198a0c4eca718e81b63f514e Mon Sep 17 00:00:00 2001
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/ossutil/use_go_mod.patch:1]: Is that an upstream patch we can fetch?

File: pkgs/tools/admin/ossutil/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  vendorSha256 = "lem9Jg4Ywv3qcIwhiZHNi1VH5HxxNr6mnefOLCzPL70=";
+
+  # don't run tests as they require secret access keys that only travis has
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = "https://github.com/aliyun/ossutil";
+    description =
+      "A user friendly command line tool to access Alibaba Cloud OSS";
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/ossutil/default.nix:23]:
```suggestion
    description = "A user friendly command line tool to access Alibaba Cloud OSS";
```

Comment (jpetrucciani) [reply pkgs/tools/admin/ossutil/default.nix:23]: Ah hmm, `nixpkgs-fmt` didn't change it for me - but yeah that looks better!

File: pkgs/tools/admin/ossutil/use_go_mod.patch (hunk 1 above)
Comment (jpetrucciani) [reply pkgs/tools/admin/ossutil/use_go_mod.patch:1]: So I got this patch from adding a commit to [my fork of that repo](https://github.com/jpetrucciani/ossutil/commit/64067e979fb24ffb198a0c4eca718e81b63f514e) (and originally I was fetching directly from that, but a friend suggested I just add the patch into my PR here) - I do plan on trying to make a PR to the original repo, however there was a [previous PR trying to make that change](https://github.com/aliyun/ossutil/pull/147/files) as well that hadn't been replied to that was no longer working (as the dependencies had changed on the main repo)

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133413` at be53eb4a run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>ossutil</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/tools/admin/ossutil/default.nix:13:15:
```
   |
13 |   patches = [ ./use_go_mod.patch ];
   |               ^
```
</li>
    <li>warning: unused-argument

Unused argument: `stdenv`.
Near pkgs/tools/admin/ossutil/default.nix:1:3:
```
  |
1 | { stdenv, lib, buildGoModule, fetchFromGitHub, fetchpatch }:
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/tools/admin/ossutil/default.nix:1:48:
```
  |
1 | { stdenv, lib, buildGoModule, fetchFromGitHub, fetchpatch }:
  |                                                ^
```
</li>
  </ul>
</details>

File: pkgs/tools/admin/ossutil/use_go_mod.patch (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/admin/ossutil/use_go_mod.patch:1]: Can you do a PR to the repo and fetch that instead? That is better for big patches to keep the repo size small.

Comment (jpetrucciani) [reply pkgs/tools/admin/ossutil/use_go_mod.patch:1]: For sure!

Comment (jpetrucciani) [reply pkgs/tools/admin/ossutil/use_go_mod.patch]: This patch file has been removed, I've opened a PR on the original repo [here](https://github.com/aliyun/ossutil/pull/155), and this now uses fetchpatch!

File: pkgs/tools/admin/ossutil/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  # this patch is required to add go mods to fetch dependencies
+  patches = [
+    (
+      fetchpatch {
+        url = "https://github.com/aliyun/ossutil/commit/64067e979fb24ffb198a0c4eca718e81b63f514e.patch";
+        sha256 = "2pn0BcbNNL+iMema54LRpG/ca5kyDugLIZQ/TMhYG/8=";
+      }
+    )
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/ossutil/default.nix:20]:
```suggestion
    (fetchpatch {
      url = "https://github.com/aliyun/ossutil/commit/64067e979fb24ffb198a0c4eca718e81b63f514e.patch";
      sha256 = "2pn0BcbNNL+iMema54LRpG/ca5kyDugLIZQ/TMhYG/8=";
    })
```

**Review by SuperSandro2000 — APPROVED** [review]
other than that

File: pkgs/tools/admin/ossutil/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/tools/admin/ossutil/default.nix:20]: nixpkgs-fmt does not hard enforce this style but you will find it in many places in nixpkgs like this.

File: pkgs/tools/admin/ossutil/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{ lib, buildGoModule, fetchFromGitHub, fetchpatch }:
+buildGoModule rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/ossutil/default.nix:2]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, fetchpatch }:

buildGoModule rec {
```

File: pkgs/tools/admin/ossutil/default.nix (hunk 3 above)
Comment (jpetrucciani) [reply pkgs/tools/admin/ossutil/default.nix]: Sounds good, I've updated it!

File: pkgs/tools/admin/ossutil/default.nix (hunk 4 above)
Comment (jpetrucciani) [reply pkgs/tools/admin/ossutil/default.nix]: updated and pushed!

**Review by benaduggan — APPROVED** [review]
LGTM! Would be super nice to get this merged soon

**Review by davidak — APPROVED** [review]
Change looks good

Builds, starts

**Pull request merged**